contracts/staking/StakingPool.sol

```diff
@@ -0,0 +1,484 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+import "../RoleAccessControl.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "./StakingPoolLib.sol";
+
+import "hardhat/console.sol";
+
+contract StakingPool is
+    Initializable,
+    RoleAccessControl,
+    ReentrancyGuard,
+    PausableUpgradeable
+{
+    struct User {
+        uint128 depositAmount;
+        uint128[5] rewardAmounts; // todo: use constant from library
```

Comment:
compiler doesnt like using a library constant here, need to add solhint ignore rule